app/src/main/java/com/example/mona/facebookoffline/EditActivity.java

```diff
@@ -41,15 +57,100 @@ public void onClick(View v) {
             public void onClick(View v) {
                 Log.d(TAG, "Clicked 'Create Post' button");
                 String msg = mPostBody.getText().toString();
-                mFacebookApiService.postMessageToPage(Constants.PAGE_ID, msg,
-                        new GraphRequest.Callback() {
-                            @Override
-                            public void onCompleted(GraphResponse response) {
-                                Log.d(TAG, "Response: " + response);
-                            }
-                        });
+                GraphRequest.Callback cb = new GraphRequest.Callback() {
+                    @Override
+                    public void onCompleted(GraphResponse response) {
+                        Log.d(TAG, "Response: " + response);
+                    }
+                };
+                if (!mPhotoUris.isEmpty()) {
+                    mFacebookApiService.publishPhotoToPage(Constants.PAGE_ID,
+                            mPhotoUris.get(0), msg, cb);
+                } else {
+                    mFacebookApiService.postMessageToPage(Constants.PAGE_ID, msg, cb);
+                }
                 finish();
             }
         });
+
+        Button addPhotosButton = (Button) findViewById(R.id.add_photos_button);
+        addPhotosButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                launchPhotoPicker();
+            }
+        });
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        Log.d(TAG, "-------------Contents of mPhotoUris---------");
+        for (Uri uri : mPhotoUris) {
+            Log.d(TAG, "Photo uri:" + uri);
+        }
+        Log.d(TAG, "-------------END---------------");
     }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        // Make sure to call the super method so that the states of our views are saved
+        super.onSaveInstanceState(outState);
+
+        outState.putSerializable(PHOTO_URI_LIST, mPhotoUris);
+    }
+
+    private void launchPhotoPicker() {
+        Intent intent = new Intent();
+        intent.setType("image/*");
+        intent.setAction(Intent.ACTION_GET_CONTENT);
+        intent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
```

Comment:
Mostly laziness. I wrote the code to select and extract the uri's from selecting multiple photos before I realized the limitation with the fb api I'm using. Rather than stripping out that code, I figure it'll be useful in the near future.